iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -442,10 +446,11 @@ public Map<String, String> getBasicStatistics(Partish partish) {
     org.apache.hadoop.hive.ql.metadata.Table hmsTable = partish.getTable();
     // For write queries where rows got modified, don't fetch from cache as values could have changed.
     Table table = getTable(hmsTable);
+    Snapshot snapshot = getSpecificSnapshot(partish.getTable(), table);
```

Comment:
should we move snapshot fetch under the `if`

Reply:
You can use hmsTable variable here instead of calling partish.getTable(); again.

Reply:
Fixed! Thanks